device.go

```diff
@@ -45,6 +45,10 @@ type DeviceDesc struct {
 
 	// Configuration information
 	Configs map[int]ConfigDesc
+
+	iSerialNumber       int   // The index into the GetStringDescriptor() to get the Serial Number
+	iProduct            int   // The index into the GetStringDescriptor() to get the Product name
+	iManufacturer       int   // The index into the GetStringDescriptor() to get the Manufacturer name
```

Comment:
in addition to the device descriptor each device has a list of different configurations, and each configuration has a number of interfaces, each with potential multiple alternate settings - see the docs for iConfiguration and iInterface fields in the USB spec.

That means for every device, apart from getting device manufacturer/product/serial, you can also ask for a text description of a configuration, and within that configuration you can ask for a description of a particular interface the configuration provides.

It makes sense to provide access to these configuration/interface string descriptors in a similar way as manufacturer/product/serial.

One way to do that would be to add a method "Description() string" to the Config and InterfaceSetting structures. That was my initial idea. It's not a great idea, because then to get the configuration description, you would need to obtain the Config/InterfaceSetting by claiming (and potentially switching) the config, which seems entirely unnecessary.

So the idea proposed in the second comment is to add methods to the Device struct, just like Manufacturer etc, for retrieving a string descriptor of a particular configuration (ConfigDescription) or of a particular interface (InterfaceDescription). To get a descriptor of an interface, you need to indicate the configuration in which this interface exists (usually 0), the number of the interface and the number of the alternate setting of that interface - only that entire triple identifies the relevant string descriptor.